packages/too-many-hooks/src/useMatchMedia.ts

```diff
@@ -0,0 +1,50 @@
+import { useCallback, useEffect, useState } from 'react'
+
+/**
+ * `useMatchMedia` hook type
+ *
+ * @export
+ * @typedef {UseMatchMedia}
+ * @param {string} query - a CSS media query to match on
+ * @param {?boolean} [default=false] - default match value
+ * @returns {boolean}
+ */
+export type UseMatchMedia = (query: string, defaultValue?: boolean) => boolean
+
+/**
+ * Returns a boolean indicating whether or not a media query matches
+ *
+ * @example
+ * To check for a screen width <= 360px
+ * ```ts
+ * const isMobile = useMatchMedia("screen and (max-width: 360px)")
+ * ```
+ * @example
+ * To check if the browser is printing the page
+ * ```ts
+ * const isPrinting = useMAtchMedia("print")
+ * ```
+ * @implements {UseMatchMedia}
+ * @param {string} query - a CSS media query to match on
+ * @param {?boolean} [default=false] - default match value
+ * @returns {boolean}
+ */
+const useMatchMedia: UseMatchMedia = (query: string, defaultValue = false): boolean => {
```

Comment:
Is there a better name for `defaultValue` here? Something like `defaultIsMatching` or `defaultMatches`?

```suggestion
const useMatchMedia: UseMatchMedia = (query: string, defaultValue = false): boolean => {
```

Reply:
Removed entirely since it now returns the entire match media list so a default doesn't make sense